ambry-rest/src/main/java/com.github.ambry.rest/PublicAccessLogHandler.java

```diff
@@ -181,4 +189,31 @@ private void logError(String msg) {
     logMessage.append(msg);
     publicAccessLogger.logError(logMessage.toString());
   }
+
+  /**
+   * If this is an SSL channel, log information about the peer certificate.
+   * @param ctx the {@link ChannelHandlerContext} for this channel.
+   */
+  private void logSSLInfo(ChannelHandlerContext ctx) {
+    if (sslLogMessage == null) {
+      sslLogMessage = new StringBuilder();
+      sslLogMessage.append("SSL (");
+      try {
+        SslHandler sslHandler = ctx.pipeline().get(SslHandler.class);
+        boolean sslUsed = sslHandler != null;
+        sslLogMessage.append("[used=").append(sslUsed).append("]");
+        if (sslUsed && publicAccessLogger.isCertLoggingEnabled()) {
+          X509Certificate certificate = (X509Certificate) sslHandler.engine().getSession().getPeerCertificates()[0];
+          X500Principal principal = certificate.getSubjectX500Principal();
+          Collection subjectAlternativeNames = certificate.getSubjectAlternativeNames();
+          sslLogMessage.append(", [principal=").append(principal).append("]");
+          sslLogMessage.append(", [san=").append(subjectAlternativeNames).append("]");
```

Comment:
Here is a sample with SSL (cert logging enabled):
```
cgetz-mn1.linkedin.biz/172.29.127.21:59634 GET /AAEAAQAAAAAAAAAAAAAAJDJhMzE4YmU5LWUzMDEtNDVlMy1hMGRlLTU2MmJlMjg4MDRjOQ.mp4, SSL ([used=true], [principal=CN=cgetz-mn1.linkedin.biz, OU=Tools, O=LinkedIn, L=Mountain View, ST=California, C=US], [san=[[2, cgetz-mn1.linkedin.biz], [6, urn:li:servicePrincipal(perf-client;None;None)], [6, urn:li:servicePrincipalMetadata(DEV;1.0.0)]]]), Request ([Host=cgetz-mn1.linkedin.biz:1175] [User-Agent=curl/7.43.0] [x-ambry-service-id=casey] [isChunked=false]), Response ([isChunked=false]), status=400, duration=58ms (chunked request receive=13ms)
```

Here is an example without SSL:
```
/172.29.127.21:59942 GET /AAEAAQAAAAAAAAAAAAAAJDJhMzE4YmU5LWUzMDEtNDVlMy1hMGRlLTU2MmJlMjg4MDRjOQ.mp4, SSL ([used=false]), Request ([Host=cgetz-mn1.linkedin.biz:1174] [User-Agent=curl/7.43.0] [x-ambry-service-id=casey] [isChunked=false]), Response ([isChunked=false]), status=400, duration=2ms (chunked request receive=1ms)
```